### DISCUSSION TITLE
Ability to specify a custom api endpoint

### DISCUSSION THREAD

**Discussion by mieliespoor** [DependencyTrack/dependency-track#2507] [Q&A] [answered]
I'm looking to deploy Dependency-Track to AWS ECS behind a single ALB. Currently I don't think this is possible as I'm unable to specify a custom oath for the api.

By default, the api lives on the path `/api`. If on the alb I specify a path for the alb to forward traffic on, it includes this path in the when the request hits the target group.

Take the following paths on the alb listener:
/api*      - api path
/*           - front end (default)

These will forward to two different target groups, but the First will then result in the full path of the api looking like `/api/api`.

So ideally, I would like to tell the api it serves on the root path and not /api path.

I need to do some more testing here, to confirm, but my initial high level tests showed this up.

Additionally, how would I work around this to have my frontend and api run on the same domain so that I don't need a second alb deployed for the api

**Comment by syalioune** [comment, marked as answer]
Hello @mieliespoor 

Your problem seems odd. Even though i'm not using ECS, I have a working setup with an EC2 instance (running docker compose) behind an ALB.
Here are my configs

- Target groups
![image](https://user-images.githubusercontent.com/6144741/222254659-a521f2a7-af22-40d7-bcf0-8f25981d47d6.png)
- ALB rules
![image](https://user-images.githubusercontent.com/6144741/222254894-c3b2ba97-483b-4302-8f93-79a37ba829e6.png)
- The `API_BASE_URL` set on the frontend side **does not include /api** : `http://dependency-track-XXXXXX.eu-west-3.elb.amazonaws.com` instead of `http://dependency-track-XXXXXXX.eu-west-3.elb.amazonaws.com/api` (that's maybe where you have a faulty setup)

**Reply by mieliespoor** [reply]
OK yes. That actually worked. I think I just had to give the task enough time to start. I'll play around with the target group health check parameters.
